pygeoapi/linked_data.py

```diff
@@ -173,104 +175,146 @@ def jsonldify_collection(cls, collection, locale_):
-    feature.pop('properties')
+def jsonldify_geometry(feature):
+    """
+        Render JSON-LD for feature with GeoJSON, Geosparql/WKT, and
+        schema geometry encodings.
+
+        :param feature: feature body to with GeoJSON geometry
+
+        :returns: None
+    """
+
+    geo = feature.get('geometry')
+    geom = asShape(geo)
+
+    # GeoJSON geometry
+    feature['geometry'] = feature.pop('geometry')
+
+    # Geosparql geometry
+    feature['geosparql:hasGeometry'] = {
+        '@type': f'http://www.opengis.net/ont/sf#{geom.geom_type}',
+        'geosparql:asWKT': {
+            '@type': 'http://www.opengis.net/ont/geosparql#wktLiteral',
+            '@value': f'{geom.wkt}'
+        }
+    }
 
-    feature['type'] = 'schema:Place'
+    # Schema geometry
+    feature['schema:geo'] = {'@type': 'schema:GeoShape'}
+    if geom.geom_type == 'Point':
+        feature['schema:geo'] = {
+            '@type': 'schema:GeoCoordinates',
+            'schema:longitude': geom.x,
+            'schema:latitude': geom.y
+        }
+        return
+
+    elif geom.geom_type == 'LineString':
+        _ = [f'{x},{y}' for (x, y) in geom.coords[:]]
+        feature['schema:geo']['schema:line'] = ' '.join(_)
+        return
+
+    elif geom.geom_type == 'MultiLineString':
+        points = list()
+        [points.extend(p.coords[:]) for p in geom.geoms]
+        _ = [f'{x},{y}' for (x, y) in points]
+        feature['schema:geo']['schema:line'] = ' '.join(_)
+        return
+
+    elif geom.geom_type == 'MultiPoint':
+        poly_geom = [(p.x, p.y) for p in geom.geoms]
+        poly_geom.append(poly_geom[0])
+
+    elif geom.geom_type == 'Polygon':
+        poly_geom = geom.exterior.coords[:]
+
+    elif geom.geom_type == 'MultiPolygon':
+        poly = unary_union(geom.buffer(0))
+        if poly.geom_type.startswith('Multi') or not poly.is_valid:
+            LOGGER.debug('Invalid Poly: {}'.format(poly.geom_type))
+            poly = poly.convex_hull
+            LOGGER.debug('New Poly: {}'.format(poly.geom_type))
+        poly_geom = poly.exterior.coords[:]
 
-    # Remove non-point geometry
-    if feature.get('geometry').get('type').lower() != 'point':
-        feature.pop('geometry')
-        geocontext = []
     else:
-        feature.get('geometry').update({
-                "lat": feature.get('geometry').get('coordinates')[1],
-                "long": feature.get('geometry').get('coordinates')[0]
-                })
-        feature.get('geometry').pop('coordinates')
-        geocontext = [{
-                        "geometry": "schema:geo",
-                        "Point": "schema:GeoCoordinates",
-                        "lat": "schema:latitude",
-                        "long": "schema:longitude"
-        }, ]
-
-    return feature, geocontext
+        poly_geom = list()
+        for p in geom.geoms:
+            try:
+                poly_geom.extend(p.coords[:])
+            except NotImplementedError:
```

Comment:
Not clear to me what's happening here? Something to do with holes?

Reply:
This is to catch `GeometryCollection` or anything else. The reduction from GeoJSON & WKT to the schema.org geometry is pretty large, and I couldn't think of a better way to reduce something like a `GeometryCollection`